Adds API docs for Saved Objects Bulk Update (#127687)
elastic/kibana: docs/api/saved-objects/bulk_update.asciidoc

[[saved-objects-api-bulk-update]]
=== Bulk update object API
++++

> <titleabbrev>Bulk update object</titleabbrev>

Reviewer: ```suggestion
<titleabbrev>Bulk update objects</titleabbrev>
```

`PUT <kibana host>:<port>/s/<space_id>/api/saved_objects/_bulk_update`

[[saved-objects-api-bulk-update-path-params]]
==== Path parameters

`space_id`::
  (Optional, string) An identifier for the space. If `space_id` is not provided in the URL, the default space is used.

[[saved-objects-api-bulk-update-request-body]]
==== Request body

`type`::
  (Required, string) Valid options include `visualization`, `dashboard`, `search`, `index-pattern`, `config`.

`id`::
  (Required, string) The object ID to update.

`attributes`::
  (Required, object) The data to persist.
+
WARNING: When you update, attributes are not validated, which allows you to pass arbitrary and ill-formed data into the API and break {kib}. Make sure any data that you send to the API is properly formed.

`references`::
  (Optional, array) Objects with `name`, `id`, and `type` properties that describe the other saved objects this object references. To refer to the other saved object, use `name` in the attributes, but never the `id`, which automatically updates during migrations or import/export.

`version`::
  (Optional, number) Ensures the version matches that of the persisted object.
  

> `namespace`:: (Optional, string) Identifier for the spaces in which to search for this object. If this is defined, it will supersede the namespace ID that is in.

Reviewer: Nit
```suggestion
`namespace`:: (Optional, string) Identifier for the space in which to update this object. If this is defined, it will supersede the space ID that is in the URL.
```

> [[saved-objects-api-bulk-update-errors-codes]]

Reviewer: Maybe this anchor is more accurate?
```suggestion
[[saved-objects-api-bulk-update-codes]]
```

==== Response code

`200`::
  Indicates a successful call. Note, this HTTP response code indicates that the bulk operation succeeded. Errors pertaining to individual
  objects will be returned in the response body. Refer to the example below for details.

[[saved-objects-api-bulk-update-example]]
==== Example

> Update existing {data-source} objects,`my-visualization`, `my-dashboard` and `my-pattern`, with a different title for each, where the id of `my-visualization` is corrupted:

Reviewer: These IDs don't reflect what is in the example. How about this?

```suggestion
Update three saved objects, where one of them does not exist:
```
